lib/lib-search/src/main/java/org/sagebionetworks/search/CloudsSearchDomainClientAdapter.java

```diff
@@ -69,14 +72,19 @@ public void sendDocuments(Iterator<Document> documents){
 						.withContentLength(batch.size());
 				UploadDocumentsResult result = client.uploadDocuments(request);
 				updateAndSendKinesisLogRecords(result.getStatus());
+				// PLFM-5570 and https://docs.aws.amazon.com/cloudsearch/latest/developerguide/limits.html
+				// limit on 1 batch per 10 seconds
+				clock.sleep(10000);
 			} catch (DocumentServiceException e) {
 				logger.error("The following documents failed to upload: " +  documentIds);
 				documentIds = null;
 				updateAndSendKinesisLogRecords(e.getStatus());
 				throw handleCloudSearchExceptions(e);
 			} catch (IOException e){
 				throw new TemporarilyUnavailableException(e);
-			}finally{
+			} catch (InterruptedException e) {
+				logger.error("sleep was interrupted");
```

Comment:
Recommend terminating on IterrupteedException.  It can occur with the graceful shutdown of an process.